tests/e2e/util.go

```diff
@@ -6295,3 +6295,141 @@ func checkClusterIdValueOnWorkloads(vs *vSphere, client clientset.Interface,
+	isPrivileged bool, command string, podName string) *v1.Pod {
+
+	podlabels := make(map[string]string)
+	podlabels["psp.vmware.com/pod-placement-opt-in"] = "true"
+	podSpec := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: ns,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:            "write-pod",
+					Image:           nginxImage,
+					SecurityContext: fpod.GenerateContainerSecurityContext(isPrivileged),
+				},
+			},
+			RestartPolicy: v1.RestartPolicyOnFailure,
+		},
+	}
+	var volumeMounts = make([]v1.VolumeMount, len(pvclaims))
+	var volumes = make([]v1.Volume, len(pvclaims))
+	for index, pvclaim := range pvclaims {
+		volumename := "data0"
+		volumeMounts[index] = v1.VolumeMount{Name: volumename, MountPath: "/data0"}
+		volumes[index] = v1.Volume{Name: volumename, VolumeSource: v1.VolumeSource{
+			PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{ClaimName: pvclaim.Name, ReadOnly: false}}}
+	}
+	podSpec.Spec.Containers[0].VolumeMounts = volumeMounts
+	podSpec.Spec.Volumes = volumes
+	if nodeSelector != nil {
+		podSpec.Spec.NodeSelector = nodeSelector
+	}
+
+	podSpec.Labels = podlabels
+	podSpec.Spec.ServiceAccountName = "default"
+	podSpec.Spec.RestartPolicy = v1.RestartPolicyAlways
+	taintKeys := []string{v1.TaintNodeNotReady, v1.TaintNodeUnreachable}
+	podSpec.Spec.Subdomain = "sample-pe-svc"
+	var x int64 = 300
+	var tolerations []v1.Toleration
+	for _, taintkey := range taintKeys {
+		var toleration v1.Toleration
+		toleration.Key = taintkey
+		toleration.Operator = v1.TolerationOpExists
+		toleration.TolerationSeconds = &x
+		toleration.Effect = v1.TaintEffectNoExecute
+		framework.Logf("toleration: %v", toleration)
+		tolerations = append(tolerations, toleration)
+	}
+	podSpec.Spec.Tolerations = tolerations
+	framework.Logf("pod spec: %v", podSpec)
+	return podSpec
+}
+
+// createVsanDPvcAndPod creates pvc and pod for vsan direct as per wffc policy.
+// It ensures that pvc and pod is in in healthy state.
+func createVsanDPvcAndPod(client clientset.Interface, ctx context.Context,
+	namespace string, sc *storagev1.StorageClass, pvcName string,
+	podName string) (*v1.PersistentVolumeClaim, *v1.Pod) {
+	framework.Logf("Creating pvc %s with storage class %s", pvcName, sc.Name)
+	pvclaim, err := fpv.CreatePVC(client, namespace,
+		getVsanDPersistentVolumeClaimSpecWithStorageClass(namespace,
+			diskSize, sc, pvcName, podName, ""))
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	ginkgo.By("Expect claim status to be in Pending state")
+	err = fpv.WaitForPersistentVolumeClaimPhase(v1.ClaimPending, client,
+		namespace, pvclaim.Name, framework.Poll, time.Minute)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred(),
+		fmt.Sprintf("Failed to find the volume in pending state with err: %v", err))
```

Comment:
remove `fmt.Sprintf()`, you don't need that here
volume -> pvc and add pvc name in the log